### DISCUSSION TITLE
How to create bin command

### DISCUSSION THREAD

**Discussion by davistsin** [callstack/react-native-builder-bob#433] [Ideas]
This library is great and helps me a lot. 👍 You all are the best in the universe.

I need to create a bin command in the library, but I added it in the project, and I didn't find this command in example/node_modules/bin.

I created the command file in the outermost src/bin, defined bin in package.json, then made the command point to the compiled file:

```
// package.json
...
  "scripts": {
    "test": "jest",
    "typecheck": "tsc --noEmit",
    "lint": "eslint \"**/*.{js,ts,tsx}\"",
    "prepack": "bob build",
    "release": "release-it --only-version",
    "example": "yarn --cwd example",
    "example-ios": "yarn example ios",
    "example-android": "yarn example android",
    "bootstrap": "yarn example && yarn install && yarn example pods",
    "clean": "del-cli android/build example/android/build example/android/app/build example/ios/build"
  },
  "bin": {
    "build-bundle": "./lib/commonjs/bin/build_bundle.js"
  },
```

Then run `yarn`, but did not see this command in example/node_modules/bin. I hope this library can support this feature.

**Comment by atlj** [comment]
Hey, thank you for the discussion. I think the main problem here would be that we are not using npm or yarn to "link" the libraries to the example app. It's all done by modifying the `babel.config.js`, `metro.config.js`, and the `react-native.config.js`. So there is no `node_modules` involved in the process.
A good solution might be to migrate the workflow into a monorepo which we are currently working on.

For a quick workaround, you can:
1. Go to `example/package.json`
2. Add a script to call your binary
```json   
"scripts": {
    "build-bundle": "node ../lib/commonjs/bin/build_bundle.js"
  },
```
3. Call `yarn build-bundle` to invoke your script.

I know this is a trivial solution but should work for the moment being.
